cumulusci/tasks/bulkdata/extract.py

```diff
@@ -269,13 +269,32 @@ def _get_mapping_for_table(self, table):
 
     def _convert_lookups_to_id(self, mapping, lookup_keys):
         """Rewrite persisted Salesforce Ids to refer to auto-PKs."""
+
+        def throw(string):  # pragma: no cover
+            raise BulkDataException(string)
+
         for lookup_key in lookup_keys:
-            lookup_info = mapping.lookups[lookup_key]
-            model = self.models[mapping.table]
-            lookup_mapping = self._get_mapping_for_table(lookup_info.table)
-            lookup_model = self.models[lookup_mapping.get_sf_id_table()]
-            key_field = lookup_info.get_lookup_key_field()
-            key_attr = getattr(model, key_field)
+            lookup_info = mapping.lookups.get(lookup_key) or throw(
+                f"Cannot find lookup info {lookup_key}"
+            )
+            model = self.models.get(mapping.table) or throw(
+                f"Cannot find model for {mapping.table}"
+            )
+
+            lookup_mapping = self._get_mapping_for_table(lookup_info.table) or throw(
+                f"Cannot find lookup mapping for {lookup_info.table}"
+            )
+
+            lookup_model = self.models.get(lookup_mapping.get_sf_id_table()) or throw(
+                f"Cannot find lookup model for {lookup_mapping}"
```

Comment:
similar question -- _create_table adds a model for each mapping's get_sf_id_table()